auth/user_mgt.go

```diff
@@ -561,6 +582,261 @@ func (c *baseClient) getUser(ctx context.Context, query *userQuery) (*UserRecord
 	return parsed.Users[0].makeUserRecord()
 }
 
+// A UserIdentifier identifies a user to be looked up.
+type UserIdentifier interface {
+	String() string
```

Comment:
I don't think this should be in the public API. Call is something like `toString()`.